go/interpreter/lfvm/memory_test.go

```diff
@@ -28,87 +28,78 @@ func TestMemory_NewMemoryIsEmpty(t *testing.T) {
 	}
 }
 
-func TestGetExpansionCosts(t *testing.T) {
-
-	tests := []struct {
+func TestGetExpansionCostsAndSize(t *testing.T) {
```

Comment:
All tests seem to assume an initial memory size of 0. There should be a test where the initial size of the memory is something larger than zero, verifying that the expansion costs only cover the difference.

This could be as simple as having two nested loops, one over the initial size, another over the target size and the use the function to make sure that allocating the full size at once has the same costs as to sum up the costs for allocating the memory in two steps. Something like:
```

func TestGetExpansionCostsAndSize_ConsistentIncrementalCharging(t *testing.T) {
	zero := uint256.NewInt(0)
	for a := uint64(0); a < 128; a += 32 {
		for b := uint64(0); b < 128; b += 32 {

			m := NewMemory()

			// Compute costs from 0 to target size.
			costA, sizeA, errA := m.getExpansionCostsAndSize(a)
			costB, sizeB, errB := m.getExpansionCostsAndSize(b)
			if errA != nil || errB != nil {
				t.Fatalf("unexpected error: %v, %v", errA, errB)
			}

			// Compute costs for increasing from size a to size b.
			ctxt := &context{gas: math.MaxInt}
			if err := m.set(zero, make([]byte, a), ctxt); err != nil {
				t.Fatalf("unexpected error: %v", err)
			}
			delta, sizeAB, err := m.getExpansionCostsAndSize(b)
			if err != nil {
				t.Fatalf("unexpected error: %v", err)
			}

			max := sizeA
			if sizeB > sizeA {
				max = sizeB
			}
			if sizeAB != max {
				t.Fatalf("size should be monotonically increasing, got: %d, %d, %d", sizeA, sizeB, sizeAB)
			}

			wantDelta := tosca.Gas(0)
			if a <= b {
				wantDelta = costB - costA
			}

			if wantDelta != delta {
				t.Fatalf("unexpected delta for expansions to %d and %d: want: %d, got: %d", a, b, wantDelta, delta)
			}
		}
	}
}
```
```

Reply:
I understand the test target, but I do not like that the test relies on utilities which are consumers of the function being tested (set). I will come out with something that test the property.